syntax/javascript.vim

```diff
@@ -337,7 +337,7 @@ if version >= 508 || !exists("did_javascript_syn_inits")
   HiLink jsSwitchColon          Noise
   HiLink jsClassMethodType      Type
   HiLink jsObjectMethodType     Type
-  HiLink jsClassDefinition      jsFuncName
+  HiLink jsClassDefinition      Constant
```

Comment:
It's actually difficult to discuss what people want. I usually judge it based on issues and PRs that show up. Truth is, almost no one ever files issues about linking groups. Occasionally we get people who file issues that they want to change the color of something, at which point we just post an example of how they can add a special color that particular group, and they are on their way.

In other words, people generally don't care what things are linked too, they just want their colors to look good.

With that in mind, my vote will always go for Vim syntax accuracy, since we always have a match group name that can then be targeted or isolated.

In regards to class stuff, we've never actually had people file an issue about this, usually it's with regards to things like parenthesis and braces and stuff.